### ISSUE TITLE
RF03 incorrectly flags lambda parameters as inconsistent table references in Databricks higher-order functions

### ISSUE THREAD

**Issue by paretho** [sqlfluff/sqlfluff#7379]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

RF03 (references.consistent) incorrectly treats lambda parameters in higher-order functions like `transform()` as table references, causing false positives when accessing nested struct fields.

Minimal example to reproduce:

```sql
-- Databricks dialect
WITH source_data AS (
  SELECT
    'test' AS id,
    '[{"taskId": {"$oid": "abc"}}]' AS json_array
)

SELECT
  id,
  transform(
    from_json(json_array, 'array<struct<taskId:struct<`$oid`:string>>>'),
    item -> struct(item.`taskId`.`$oid` AS task_id)
  ) AS transformed
FROM source_data
```

## Configuration

```ini
[sqlfluff]
dialect = databricks
```

## SQLFluff Version

```
sqlfluff, version 3.4.1
```

## Notes

A similar fix was made for RF02 with Snowflake lambda parameters in #6689. RF03 likely needs the same treatment to recognize that lambda parameters are a separate scope from table/CTE references.

### Expected Behaviour

No RF03 violation. The `item.` prefix is required syntax to access fields from the lambda parameter - it's not a table qualification choice.

### Observed Behaviour

```
L:  10 | P:   3 | RF03 | Unqualified reference 'id' found in single table select.
L:  12 | P:  15 | RF03 | Unqualified reference 'json_array' found in single table select.
L:  13 | P:  20 | RF03 | Qualified reference 'item.`taskId`.`$oid`' found in single table select which is inconsistent with previous references.
```

### How to reproduce

Simply run sqlfluff on the attached example.

[rf03_lambda_example.sql](https://github.com/user-attachments/files/24715385/rf03_lambda_example.sql)

### Dialect

databricks

### Version

3.4.1

### Configuration

[sqlfluff]
templater = dbt
dialect = databricks
max_line_length = 120

[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = .

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
